L1Trigger/L1TTrackMatch/plugins/L1TrackJetProducer.cc

```diff
@@ -63,73 +65,73 @@ class L1TrackJetProducer : public stream::EDProducer<> {
-      nStubs4DisplacedChi2_((float)iConfig.getParameter<double>("nStubs4DisplacedChi2")),
-      nStubs5DisplacedChi2_((float)iConfig.getParameter<double>("nStubs5DisplacedChi2")),
-      nStubs4DisplacedBend_((float)iConfig.getParameter<double>("nStubs4DisplacedBend")),
-      nStubs5DisplacedBend_((float)iConfig.getParameter<double>("nStubs5DisplacedBend")),
-      nDisplacedTracks_((int)iConfig.getParameter<int>("nDisplacedTracks")),
-      dzPVTrk_((float)iConfig.getParameter<double>("MaxDzTrackPV")),
-      tTopoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>(edm::ESInputTag("", ""))),
+    : tTopoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>(edm::ESInputTag("", ""))),
       trackToken_(consumes<vector<TTTrack<Ref_Phase2TrackerDigi_>>>(iConfig.getParameter<InputTag>("L1TrackInputTag"))),
-      PVtxToken_(consumes<vector<l1t::Vertex>>(iConfig.getParameter<InputTag>("L1PVertexCollection"))) {
+      PVtxToken_(consumes<l1t::VertexWordCollection>(iConfig.getParameter<InputTag>("VertexInputTag"))) {
+  trkZMax_ = (float)iConfig.getParameter<double>("trk_zMax");
+  trkPtMax_ = (float)iConfig.getParameter<double>("trk_ptMax");
+  trkPtMin_ = (float)iConfig.getParameter<double>("trk_ptMin");
+  trkEtaMax_ = (float)iConfig.getParameter<double>("trk_etaMax");
+  nStubs4PromptChi2_ = (float)iConfig.getParameter<double>("nStubs4PromptChi2");
+  nStubs5PromptChi2_ = (float)iConfig.getParameter<double>("nStubs5PromptChi2");
+  nStubs4PromptBend_ = (float)iConfig.getParameter<double>("nStubs4PromptBend");
+  nStubs5PromptBend_ = (float)iConfig.getParameter<double>("nStubs5PromptBend");
+  trkNPSStubMin_ = (int)iConfig.getParameter<int>("trk_nPSStubMin");
+  minTrkJetpT_ = iConfig.getParameter<double>("minTrkJetpT");
+  etaBins_ = (int)iConfig.getParameter<int>("etaBins");
+  phiBins_ = (int)iConfig.getParameter<int>("phiBins");
+  zBins_ = (int)iConfig.getParameter<int>("zBins");
+  d0CutNStubs4_ = (float)iConfig.getParameter<double>("d0_cutNStubs4");
+  d0CutNStubs5_ = (float)iConfig.getParameter<double>("d0_cutNStubs5");
+  lowpTJetMinTrackMultiplicity_ = (int)iConfig.getParameter<int>("lowpTJetMinTrackMultiplicity");
+  lowpTJetThreshold_ = (float)iConfig.getParameter<double>("lowpTJetThreshold");
+  highpTJetMinTrackMultiplicity_ = (int)iConfig.getParameter<int>("highpTJetMinTrackMultiplicity");
+  highpTJetThreshold_ = (float)iConfig.getParameter<double>("highpTJetThreshold");
+  displaced_ = iConfig.getParameter<bool>("displaced");
+  nStubs4DisplacedChi2_ = (float)iConfig.getParameter<double>("nStubs4DisplacedChi2");
+  nStubs5DisplacedChi2_ = (float)iConfig.getParameter<double>("nStubs5DisplacedChi2");
+  nStubs4DisplacedBend_ = (float)iConfig.getParameter<double>("nStubs4DisplacedBend");
+  nStubs5DisplacedBend_ = (float)iConfig.getParameter<double>("nStubs5DisplacedBend");
+  nDisplacedTracks_ = (int)iConfig.getParameter<int>("nDisplacedTracks");
+  dzPVTrk_ = (float)iConfig.getParameter<double>("MaxDzTrackPV");
```

Comment:
I'm confused by the casts here. Why are the `double`s casted to `float`? The `_cfi.py` explicitly provides `double` values, could those just be `float`s in the `_cfi`? Why are the `int`s recast back into `int`? Is there something I am missing here?

Reply:
Ah, again, thank you.

Reply:
Written that, I agree the casts look confusing (and are technically unnecessary). Although I see the casts were there already before. But then I could ask why the member initialization was moved away from member initializer list? The member initializer list is recommended where feasible.

Reply:
I put them there in order to avoid remembering the order. But I transferred them back both in simulation and emulation. And removed the casts